osquery/filesystem/filesystem.cpp

```diff
@@ -107,6 +107,67 @@ struct OpenReadableFile : private boost::noncopyable {
   bool blocking_io;
 };
 
+ExpectedTextFile readTextFile(const std::string& path,
```

Comment:
What's the goal with this new file reading method? Is there a limitation with the existing methods?

Reply:
Heads up, still interested in this question @alessandrogario :D

Reply:
> Heads up, still interested in this question @alessandrogario :D

The current readFile() methods that we have under filesystem.cpp are pretty big and rather complex; an additional issue was that they don't return many details when something fails - just a text string.

This was an attempt to use a simpler function, with better error handling and the more modern `Expectd<ValueType, ErrorType>` way of handling returned values that was implemented in the experimental branch

Reply:
Do you anticipate refactoring to eliminate the old method in the codebase? I'd like to avoid the proliferation of different functions for achieving the same goal.

Reply:
> Do you anticipate refactoring to eliminate the old method in the codebase? I'd like to avoid the proliferation of different functions for achieving the same goal.

I think it would be useful to refactor the filesystem functions so that they become smaller and easier to work with; something similar to the fs interface we have in BPF (`osquery/events/linux/bpf/ifilesystem.h`) could prove really useful when implementing tests, because it lets us swap the implementation with a mocked class. An example for this usage can be also found in the BPF code: `osquery/events/tests/linux/bpf/mockedfilesystem.cpp`